examples/seismic/skew_self_adjoint/operators.py

```diff
@@ -0,0 +1,330 @@
+
+    Returns
+    ----------
+    The time update stencil
+    """
+    # Get the Functions for buoyancy, velocity, and wOverQ
+    assert 'b' in model, "model dictionary must contain key 'b'"
+    assert 'v' in model, "model dictionary must contain key 'b'"
+    assert 'wOverQ' in model, "model dictionary must contain key 'wOverQ'"
+    b = model['b']
+    v = model['v']
+    wOverQ = model['wOverQ']
+
+    # Define time step of pressure wavefield to be updated
+    forward = kwargs.get('forward', True)
+
+    if forward:
+        field_next = field.forward
+        field_prev = field.backward
+    else:
+        field_next = field.backward
+        field_prev = field.forward
+
+    # Get the source
+    q = kwargs.get('q', 0)
+
+    # Define the time update equation for 2d/3d
+    # TODO: consider replacing conditional logic with @mloubout's comprehension
+    # space_fd = sum([getattr(b * getattr(field, 'd%s'%d.name)(x0=d+d.spacing/2)),
+    #            'd%s'%d.name)(x0=d-d.spacing/2)) for d in field.dimensions[1:]])
+    if len(field.data.shape) == 3:
+        t, x, z = field.dimensions
+        eq_time_update = (t.spacing**2 * v**2 / b) * \
+            ((b * field.dx(x0=x+x.spacing/2)).dx(x0=x-x.spacing/2) +
+             (b * field.dz(x0=z+z.spacing/2)).dz(x0=z-z.spacing/2) + q) + \
+            (2 - t.spacing * wOverQ) * field + \
+            (t.spacing * wOverQ - 1) * field_prev
+
+    else:
+        t, x, y, z = field.dimensions
+        eq_time_update = (t.spacing**2 * v**2 / b) * \
+            ((b * field.dx(x0=x+x.spacing/2)).dx(x0=x-x.spacing/2) +
+             (b * field.dy(x0=y+y.spacing/2)).dy(x0=y-y.spacing/2) +
+             (b * field.dz(x0=z+z.spacing/2)).dz(x0=z-z.spacing/2) + q) + \
+            (2 - t.spacing * wOverQ) * field + \
+            (t.spacing * wOverQ - 1) * field_prev
+
+    return [Eq(field_next, eq_time_update)]
+
+
+def ISO_FwdOperator(model, src, rec, time_axis, space_order=8, save=False, **kwargs):
+    """
+    Construct a forward modeling operator in a variable density visco- acoustic media.
+    See implementation notebook ssa_01_iso_implementation1.ipynb for more details.
+
+    Parameters
+    ----------
+    model : Dictionary <string>:<Function>, contains:
+        'b': Buoyancy = reciprocal density (units: m^3/kg)
+        'v': Velocity (units: m/msec or km/sec)
+        'wOverQ': The w/Q field for dissipation only attenuation.
+    src : SparseTimeFunction (PointSource)
+        Source position and time signature.
+    rec : SparseTimeFunction (PointSource)
+        Receiver positions and time siganture.
+    time_axis : TimeAxis
+        Defines temporal sampling
+    space_order : int, optional, Defaults to 8
+        Space discretization order.
+    save : int or Buffer, optional
+        Saving flag, True saves all time steps. False saves three timesteps.
+        Defaults to False.
+
+    Returns
+    ----------
+    The operator implementing forward modeling
```

Comment:
Some nitpicking: 
* operator -> Operator
* full stop at the end

Reply:
assuming "full stop" means punctuation, done